File: src/main/java/Lv2/_54_2개_이하로_다른_비트/Solution.java
package Lv2._54_2개_이하로_다른_비트;

import java.util.Arrays;

class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        for(int i=0; i<numbers.length; i++) {
            long num = numbers[i];
            String bNum = Long.toBinaryString(numbers[i]);
            if(num%2==0) {
                answer[i] = num+1;
            }else{
                int idx = bNum.lastIndexOf("0");

                if (idx == -1) {
                    String tmp = "10" + bNum.substring(1, bNum.length());
                    answer[i] = Long.parseLong(tmp, 2);
                } else {
                    String tmp = bNum.substring(0, idx) + "10" + bNum.substring(idx + 2, bNum.length());
                    answer[i] = Long.parseLong(tmp, 2);
                }
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(Arrays.toString(s.solution(new long[] {2, 7})));
    }
}

File: src/main/java/Lv2/_54_2개_이하로_다른_비트/Solution.js
function solution(numbers) {
    var answer = [];

    for(let i=0; i<numbers.length; i++) {
        let num = numbers[i];
        if(!(num % 2)) {
            answer[i] = num+1;
        }else{
            let bin = num.toString(2);
            let unit = cal(bin);
            answer[i] = unit;
        }
    }

    return answer;
}

function cal (num) {
    let addNum = "10";
    let maxLen = num.length;
    let index = -1;
    for(let i=maxLen; i>0; i--) {
        // 우측에서 첫번째 0 찾기
        if(num[i]==='0') {
            index = i;
            break;
        }
    }

    if(index == -1) {
        let rnum = addNum + num.substring(1, maxLen);
        return parseInt(rnum, 2);
    }else {
        let rnum = num.substring(0, index) + addNum + num.substring(index+2, maxLen);
        return parseInt(rnum, 2);
    }

}

console.log(solution([2,127,129]));